src/nautilus/cpu.c

```diff
@@ -222,7 +222,10 @@ ulong_t
 nk_detect_cpu_freq (uint32_t cpu) 
 {
     uint8_t flags = irq_disable_save();
-    ulong_t khz = i8254_calib_tsc();
+    ulong_t khz = ULONG_MAX;
```

Comment:
on riscv, you can get this using the timebase of the cpu's node. It's hard to get it but you technically can